### PR TITLE
[2.9] Move Upgrade Test Configurations and Load Functions

### PR DESCRIPTION
**Pull request by caliskanugur**
## Problem

Upgrade configuration and its load functions are the only configurations left in the validation packages.
 
## Solution

Move upgrade configuration and its load functions to Shepherd: https://github.com/rancher/shepherd/pull/28

### Automated Testing

* Test types added/modified:
    * Validation (Go Framework)

### CODE REVIEW

File: go.mod
Comment (floatingman) [inline go.mod:426]: Should this be in here?

Comment (caliskanugur) [reply go.mod:426]: That's a great question. Yes, it shouldn't be there I'll change it shortly while doing sanity checks, once Shepherd is merged, I'll update it again.

**Review by slickwarren — COMMENTED** [review]
lgtm - pending review of jenkins jobs

**Comment by caliskanugur** [comment]
Backports:

2.7: https://github.com/rancher/rancher/pull/44431
2.8: https://github.com/rancher/rancher/pull/44430

**Review by igomez06 — APPROVED** [review]
LGTM

**Review by floatingman — APPROVED** [review]

**Review by markusewalker — APPROVED** [review]

**Pull request merged**
